hack/README.md

```diff
@@ -43,6 +43,38 @@ thus the recipe has separate 'functions' for each phase of running KPNG.
 - build: compile kpng and push it to a registry
 - install: delete the kpng daemonset and redeploy it
 
+# E2E - Testing a single scenario of failure
+As soon the `./hack/test_e2e.sh -i IP_FAMILY -b BACKEND` finish, it will create a kubernetes cluster based in `kind + kpng`.  
+This cluster will keep running until the developer decide to manually remove it with all files generated in the kpng tree.
+
+*<ins>Files generated by test_e2e.sh in the kpng tree</ins>*:
+| Filename                | Description                                  |
+|-------------------------|----------------------------------------------|
+| artifacts               | Artifacts from the tests                     |
+| e2e.test                | The generated e2e.test                       |
+| ginkgo                  | Ginkgo binary                                |
+| kubeconfig_tests.conf   | The kubeconfig from the created kind cluster |
+
+However, these files can be useful for developers speed up and execute a `single failure test` from the E2E framework.  
+
+The example below focus **only** in the test `should be able to preserve UDP traffic when server pod cycles for a ClusterIP service` and **skip**: `Feature|Federation|PerformanceDNS|Disruptive|Serial|LoadBalancer|KubeProxy|GCE|Netpol|NetworkPolicy` tests.
+
+*<ins>Executing a single KPNG test</ins>*:
+```
+kpng> ./hack/test_e2e.sh -i ipv4 -b iptables
+kpng> cd hack/temp/e2e
+kpng/e2e2>./ginkgo --nodes="25" \
+       --focus="should be able to preserve UDP traffic when server pod cycles for a ClusterIP service" \
+       --skip="Feature|Federation|PerformanceDNS|Disruptive|Serial|LoadBalancer|KubeProxy|GCE|Netpol|NetworkPolicy" \
+        ./e2e.test \
+        -- \
+        --kubeconfig="kubeconfig_tests.conf" \
+        --provider="local" \
+        --dump-logs-on-failure=false \
+        --report-dir="artifacts/reports" \
```

Comment:
maybe you could explain what is expected in the reports and how to find things.. onliner explaining the file and utility would be great.